sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpTest.java

```diff
@@ -53,4 +54,17 @@ public void shouldGetCustomFileSuffixes() {
     assertThat(csharp.getFileSuffixes()).containsOnly(".cs", ".csharp");
   }
 
+  @Test
+  public void equals_and_hashCode_considers_configuration() {
+    MapSettings otherSettings = new MapSettings();
+    otherSettings.setProperty("key", "value");
+    CSharp otherCSharp = new CSharp(otherSettings.asConfig());
+    CSharp sameCSharp = new CSharp(settings.asConfig());
+
+    assertThat(csharp).isEqualTo(sameCSharp)
```

Comment:
can you add a test comparing it to a VBNet?

Reply:
Only with fake VbNet class, because it's in separate library.